submission/Vagrantfile

```diff
@@ -6,9 +6,7 @@ Vagrant.configure("2") do |config|
     end
 
     config.nfs.functional = false
-    config.vm.provision 'shell', inline: 'mv /vagrant/submission.zip .; \
-                                            unzip submission.zip; \
-                                            chown -R vagrant:vagrant .'
+    config.vm.provision 'shell', inline: 'cd /vagrant; chmod +x script.sh; unzip archive.zip'
```

Comment:
You can `chmod +x script.sh` in `launch.sh` before rsync, so that rsync will send the file with the right permissions. And let the checker script perform the unzip, it's their payload to worry about, we can be lazy here.